internal/lib/stats/metrics.go

```diff
@@ -0,0 +1,445 @@
+			}, {
+				Name:      "container_spec_cpu_shares",
+				Help:      "CPU share of the container.",
+				LabelKeys: baseLabelKeys,
+			},
+		},
+		NetworkMetrics: {
+			{
+				Name:      "container_network_receive_bytes_total",
+				Help:      "Cumulative count of bytes received",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_receive_packets_total",
+				Help:      "Cumulative count of packets received",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_receive_packets_dropped_total",
+				Help:      "Cumulative count of packets dropped while receiving",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_receive_errors_total",
+				Help:      "Cumulative count of errors encountered while receiving",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_bytes_total",
+				Help:      "Cumulative count of bytes transmitted",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_packets_total",
+				Help:      "Cumulative count of packets transmitted",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_packets_dropped_total",
+				Help:      "Cumulative count of packets dropped while transmitting",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_errors_total",
+				Help:      "Cumulative count of errors encountered while transmitting",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			},
+		},
+		"oom": {
+			{
+				Name:      "container_oom_events_total",
+				Help:      "Count of out of memory events observed for the container",
+				LabelKeys: baseLabelKeys,
+			},
+		},
+		"processes": {
+			{
+				Name:      "container_processes",
+				Help:      "Number of processes running inside the container.",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_file_descriptors",
+				Help:      "Number of open file descriptors for the container.",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_sockets",
+				Help:      "Number of open sockets for the container.",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_threads_max",
+				Help:      "Maximum number of threads allowed inside the container, infinity if value is zero",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_threads",
+				Help:      "Number of threads running inside the container",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_ulimits_soft",
+				Help:      "Soft ulimit values for the container root process. Unlimited if -1, except priority and nice",
+				LabelKeys: append(baseLabelKeys, "ulimit"),
+			},
+		},
+	}
+	return descriptorsMap
+}
+
+func sandboxBaseLabelValues(sb *sandbox.Sandbox) []string {
+	// TODO FIXME: image?
```

Comment:
(I wrote this TODO) What does cadvisor report as the image for the POD container?

Reply:
I tried to look it up but couldn't find what they are using exactly as image name for POD containers. They seem to use the image name from the container info ([here](https://github.com/google/cadvisor/blob/570d8a7eea0a7669d0a1d024190010a200bbef7b/metrics/prometheus.go#L1782)  and [here](https://github.com/google/cadvisor/blob/570d8a7eea0a7669d0a1d024190010a200bbef7b/metrics/prometheus.go#L1807) but not sure how they differentiate between a container and a pod, and what image name exactly they are using for pods. 

my suggestion would be using the sandbox name defined in the sandbox metadata (e.g. [here](https://github.com/cri-o/cri-o/blob/345b566b48ddd45d400f7a14fff5bde7caa6ab30/test/testdata/sandbox_config.json#L3))